src/components/menuFilter/menu-filter.spec.js

```diff
@@ -0,0 +1,128 @@
+
+  it('should set the correct tabindex if specified', function() {
+    var filterMenuInput = setupFilterMenu(defaultAttrs + 'tabindex="3"').find('input');
+    expect(filterMenuInput.attr('tabindex')).toBe('3');
+  });
+
+  it('should show clear search button if search term exists', function() {
+    scopeWithController.ctrl.searchTerm = 'cool beans';
+
+    var filterMenuClearButton = setupFilterMenu().find('button');
+
+    expect(filterMenuClearButton).not.toBe(null);
+  });
+
+  it('should clear search term with clear button', function() {
+    scopeWithController.ctrl.searchTerm = 'cool beans';
+
+    var filterMenuClearButton = setupFilterMenu().find('button');
+
+    filterMenuClearButton.triggerHandler('click');
+    scopeWithController.$digest();
+
+    expect(scopeWithController.ctrl.searchTerm).toBe('');
+  });
+
+  it('should filter the items depending on substring matching', function() {
+    var filterMenu = setupFilterMenu();
+
+    expect(scopeWithController.ctrl.filteredItems.length).toBe(3);
+
+    scopeWithController.ctrl.searchTerm = 'on';
+    scopeWithController.$digest();
+
+    expect(scopeWithController.ctrl.filteredItems.length).toBe(1);
+    expect(scopeWithController.ctrl.filteredItems[0]).toBe('one');
+  });
+
+  it('defaults to the aria-label to the placeholder text if none specified', function() {
+    var filterMenuInput = setupFilterMenu(defaultAttrs + 'placeholder="hello world"').find('input');
+    expect(filterMenuInput.attr('placeholder')).toBe('hello world');
+  });
+
+  it('should preserve existing aria-label', function() {
+    var filterMenu = setupFilterMenu(defaultAttrs + 'aria-label="angular rocks"');
+    expect(filterMenu.attr('aria-label')).toBe('angular rocks');
+  });
+
+  it('should expect an aria-label if none is present', inject(function($compile, $rootScope, $log) {
+    spyOn($log, 'warn');
+    setupFilterMenu();
+    $rootScope.$apply();
+    expect($log.warn).toHaveBeenCalled();
+    $log.warn.calls.reset();
+    setupFilterMenu(defaultAttrs + 'aria-label="angular rocks"');
+    $rootScope.$apply();
+    expect($log.warn).not.toHaveBeenCalled();
+  }));
+
+
+  it('should should stop keydown propagation', function() {
+    var filterMenu = setupFilterMenu();
+
+    var keydownEvent = {
+      keyCode: 22,
+      type: 'keydown',
+      target: filterMenu,
+      stopPropagation: jasmine.createSpy()
+    };
+
+    filterMenu.triggerHandler(keydownEvent);
+
+    expect(keydownEvent.stopPropagation).toHaveBeenCalled();
+  });
+
+  it('should should allow esc to propagate', function() {
+    var filterMenu = setupFilterMenu();
+
+    var keydownEvent = {
+      keyCode: 27,
+      type: 'keydown',
+      target: filterMenu,
+      stopPropagation: jasmine.createSpy()
+    };
+
+    filterMenu.triggerHandler(keydownEvent);
+
+    expect(keydownEvent.stopPropagation).not.toHaveBeenCalled();
+  });
+
+
+  function setupFilterMenu(attrs, scope) {
+    inject(function($compile, $rootScope) {
```

Comment:
Instead of injecting here, inject these in the beforeEach, save them to variables and use them here.